server/core/src/irodsReServer.cpp

```diff
@@ -77,6 +78,30 @@ namespace {
         }
     }
 
+    void load_available_executors(){
+        auto exec = irods::get_server_property<std::unordered_map<std::string, int> >(irods::HOSTS_AVAILABLE_RULE_EXECUTORS);
+        for(const auto &[hostname, port] : exec){
+            available_hosts.push_back(std::make_pair(hostname, port));
+        }
+        rodsEnv env;
+        _getRodsEnv(env);
+        available_hosts.push_back(std::make_pair(env.rodsHost,
+                                                 env.rodsPort));
+    }
+    ix::client_connection get_next_connection(){
+        static std::atomic<int> host_index = 0;
+        if(available_hosts.empty())
+            load_available_executors();
+        int cur = (host_index++) % available_hosts.size();
```

Comment:
is this random enough?

Reply:
do we want/need randomness?

round robin would guarantee that the delay server distributes work evenly across the executors. it would be more deterministic too.

Reply:
yes?    why do we want determinism?  what happens when someone takes one out of the list?   hmm... more thoughts...

Reply:
it being deterministic isn't that important, but it can help with testing. i don't see randomness helping much here.

the delay server doesn't see changes to the config file right now. if it did respond to changes to the list of executors, i'd expect the delay server to pick the next executor and attempt to connect and execute the rule.